public/app/features/alerting/unified/components/notification-policies/Filters.tsx

```diff
@@ -0,0 +1,172 @@
+import { Button, Field, Icon, Input, Label as LabelElement, Select, Tooltip } from '@grafana/ui';
+import { Receiver, RouteWithID } from 'app/plugins/datasource/alertmanager/types';
+
+import { useURLSearchParams } from '../../hooks/useURLSearchParams';
+import { parseMatchers } from '../../utils/alertmanager';
+import { Label } from '../../utils/notification-policies';
+
+interface NotificationPoliciesFilterProps {
+  receivers: Receiver[];
+  onChangeLabels: (labels: Label[] | undefined) => void;
+  onChangeReceiver: (receiver: string | undefined) => void;
+}
+
+const NotificationPoliciesFilter: FC<NotificationPoliciesFilterProps> = ({
+  receivers,
+  onChangeReceiver,
+  onChangeLabels,
+}) => {
+  const [searchParams, setSearchParams] = useURLSearchParams();
+  const searchInputRef = useRef<HTMLInputElement | null>(null);
+  const { queryString, contactPoint } = getNotificationPoliciesFilters(searchParams);
+
+  useEffect(() => {
+    onChangeReceiver(contactPoint);
+  }, [contactPoint, onChangeReceiver]);
+
+  useEffect(() => {
+    if (queryString) {
+      const labels: Label[] = parseMatchers(queryString ?? '').map(({ name, value }) => [name, value]);
+      onChangeLabels(labels.length > 0 ? labels : undefined);
+    } else {
+      onChangeLabels(undefined);
+    }
+  }, [onChangeLabels, queryString]);
+
+  const clearFilters = useCallback(() => {
+    if (searchInputRef.current) {
+      searchInputRef.current.value = '';
+    }
+    setSearchParams({ contactPoint: undefined, queryString: undefined });
+  }, [setSearchParams]);
+
+  const receiverOptions: Array<SelectableValue<string>> = receivers.map(toOption);
+  const selectedContactPoint = receiverOptions.find((option) => option.value === contactPoint) ?? null;
+
+  const hasFilters = queryString || contactPoint;
+
+  return (
+    <Stack direction="row" alignItems="flex-end" gap={0.5}>
+      <Field
+        style={{ marginBottom: 0 }}
+        label={
+          <LabelElement>
+            <Stack gap={0.5}>
+              <span>Search by labels</span>
+              <Tooltip
+                content={
+                  <div>
+                    Find notification policies that match the given set of labels, ex:
+                    <code>{`severity=critical, team=operations`}</code>
+                  </div>
+                }
+              >
+                <Icon name="info-circle" size="sm" />
+              </Tooltip>
+            </Stack>
+          </LabelElement>
+        }
+      >
+        <Input
+          ref={searchInputRef}
+          placeholder="Search"
+          data-testid="search-query-input"
+          prefix={<Icon name="search" />}
+          onChange={(event) => {
+            setSearchParams({ queryString: event.currentTarget.value });
+          }}
+          defaultValue={queryString}
+          width={46}
+        />
+      </Field>
+      <Field label="Search by contact point" style={{ marginBottom: 0 }}>
```

Comment:
Shouldn't it have its own class instead of using an inline style?

Reply:
Yea it should, will update!